src/briefcase/platforms/web/static.py

```diff
@@ -0,0 +1,347 @@
+                with briefcase_css_path.open("a", encoding="utf-8") as css_file:
+                    self._process_wheel(wheelfile, css_file=css_file)
+
+        return {}
+
+
+class HTTPHandler(SimpleHTTPRequestHandler):
+    """Convert any HTTP request into a path request on the static content
+    folder."""
+
+    def translate_path(self, path):
+        return str(self.server.base_path / path[1:])
+
+
+class LocalHTTPServer(HTTPServer):
+    """A HTTP server that serves local static content."""
+
+    def __init__(self, base_path, host, port, RequestHandlerClass=HTTPHandler):
+        self.base_path = base_path
+        super().__init__((host, port), RequestHandlerClass)
+
+
+class StaticWebRunCommand(StaticWebMixin, RunCommand):
+    description = "Run a static web project."
+
+    def add_options(self, parser):
+        super().add_options(parser)
+        parser.add_argument(
+            "--host",
+            default="localhost",
+            help="The host on which to run the server (default: localhost)",
+            required=False,
+        )
+        parser.add_argument(
+            "-p",
+            "--port",
+            default=8080,
+            type=int,
+            help="The port on which to run the server (default: 8080)",
+            required=False,
+        )
+        parser.add_argument(
+            "--no-browser",
+            action="store_false",
+            dest="open_browser",
+            help="Don't open a web browser on the newly opened server.",
+            required=False,
+        )
+
+    def run_app(self, app: AppConfig, host, port, open_browser, **kwargs):
+        """Start the application.
+
+        :param app: The config object for the app
+        :param host: The host on which to run the server
+        :param port: The device UDID to target. If ``None``, the user will
+            be asked to select a device at runtime.
+        :param open_browser: Should a browser be opened on the newly started
+            server.
+        """
+        self.logger.info(
+            f"Starting web server on http://{host}:{port}/",
+            prefix=app.app_name,
+        )
+
+        httpd = None
+        try:
+            # Create a local HTTP server
+            httpd = LocalHTTPServer(
+                self.project_path(app),
+                host=host,
+                port=port,
+            )
+
+            # If requested, open a brower tab on the newly opened server.
+            if open_browser:
+                webbrowser.open_new_tab(f"http://{host}:{port}")
+
+            self.logger.info(
+                "Web server log output (type CTRL-C to stop log)...",
+                prefix=app.app_name,
+            )
+            self.logger.info("=" * 75)
+
+            # Run the server.
+            httpd.serve_forever()
+        except KeyboardInterrupt:
+            # CTRL-C is the accepted way to stop the server.
+            pass
```

Comment:
This brings up a point I considered when I updated the output streamer to re-raise `KeyboardInterrupt`; that is, when we tell the user to press `CTRL+C`, should we also show them the `Aborted by User` message when Briefcase exits?

This suggests so; I can create a separate PR to imitate this for Android and iOS.

Reply:
Agreed - we should only log a message when something is unexpected. In this (and the Android/iOS case), Ctrl-C is normal shutdown behavior.